src/pages/index.js

```diff
@@ -23,56 +24,109 @@ import {
   profileTitle,
   profileDescriptionInput,
   profileDescription,
+  profileImage,
   placeTitleInput,
   imageURLInput,
   validationSettings,
 } from "../utils/constant.js";
 
-const addCardPopup = new PopupWithForm("#place-add-modal", (cardData) => {
-  const card = createCard(cardData);
-  section.addItem(card);
-  addCardPopup.close();
-  addFormValidator.disableSubmitBtn();
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "e1bd6119-1778-4022-8116-9e8cc4e0d599",
+    "content-type": "application/json",
+  },
 });
-addCardPopup.setEventListeners();
 
 const userInfo = new UserInfo({
   nameSelector: profileTitle,
   jobSelector: profileDescription,
+  imageSelector: profileImage,
+});
+
+const editAvatarPopup = new PopupWithForm("#modal-edit-avatar", (data) => {
+  return api
+    .updateAvatar(data)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+});
+editAvatarPopup.setEventListeners();
+
+document
+  .querySelector("#profile-image-pencil")
+  .addEventListener("click", () => {
+    editAvatarPopup.open();
+  });
+
+const confirmAction = (card, cardId) => {
+  return api.deleteCard(cardId).then(() => {
+    card.removeCard();
+  });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**